win32ss/user/ntuser/message.c

```diff
@@ -12,9 +12,33 @@
 DBG_DEFAULT_CHANNEL(UserMsg);
 
 #define PM_BADMSGFLAGS ~((QS_RAWINPUT << 16)|PM_QS_SENDMESSAGE|PM_QS_PAINT|PM_QS_POSTMESSAGE|PM_QS_INPUT|PM_NOYIELD|PM_REMOVE)
+#define EXCEPTION_ACCESS_VIOLATION ((DWORD)0xC0000005)
+
+/* Strings that are OK to pass between user and kernel mode
+ * There may be other strings needed that can easily be added here. */
+WCHAR StrUserKernel[3][20] = {{L"intl"}, {L"Environment"}, {L"Policy"}};
 
 /* FUNCTIONS *****************************************************************/
 
+/* PosInArray checks for strings that can pass between user and kernel mode.
+ * See: https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-settingchange
+ * It mentions 'Environment', 'intl', and 'Policy'.
+ * These strings are enumerated in the StrUserKernel definition.
+ * Returns: A positive integer indicating its position in the array if the
+ * string is found, or returns a minus one (-1) if the string is not found. */
+INT PosInArray(WCHAR String[])
+{
+    INT i;
+    INT End = ARRAYSIZE(StrUserKernel);
+
+    for (i = 0; i < End; ++i)
+    {
+        if (wcsncmp(String, StrUserKernel[i], sizeof(StrUserKernel[0])) == 0)
```

Comment:
```suggestion
        if (wcsncmp(String, StrUserKernel[i], _countof(StrUserKernel[0])) == 0)
```

Reply:
This only checks the first 4 chars of each string indeed.

Reply:
> Why `sizeof(StrUserKernel[0])` ? since the wcsncmp takes a number of characters (and not a number of bytes)

@HBelusca, I will correct this, I missed the fact that wcsncmp takes characters and not bytes. Thanks.

Reply:
Changed ```sizeof``` to ```_countof```. Thanks.

Reply:
Right, it tries to use `20` for every string, even strings that are smaller.
Why is that used as max length?

Reply:
@learn-more, I wanted to make sure that we had enough space to handle the 3 specific values included here and the longest was 'environment'. This requires 12 characters when including the ending NULL. So I arbitrarily added a bit more length to try and take care of some future cases that might be longer and rounded up to a multiple of 10. I included the extra two (2) values to show how this testing methodology could be easily extended in the future.